File: Easy/13. Roman to Integer/RomanToInteger.java
/*
LeetCode Easy: 13. Roman to Integer
https://leetcode.com/problems/roman-to-integer/

Check RomanToInteger.cpp for brief solution explanation.
*/
class Solution {
    public int romanToInt(String s) {
        HashMap<Character, Integer>legend = new HashMap<Character, Integer>();
        legend.put('I', 1);
        legend.put('V', 5);
        legend.put('X', 10);
        legend.put('L', 50);
        legend.put('C', 100);
        legend.put('D', 500);
        legend.put('M', 1000);
        int pre_value = 10000, ans = 0;
        for(int i = 0;i<s.length();i++){
            int cur_value = legend.get(s.charAt(i));
            if(pre_value<cur_value){
                ans-=2*pre_value;
            }
            ans+=cur_value;
            pre_value = cur_value;
        }
        return ans;
    }
}

File: Easy/13. Roman to Integer/RomanToInteger.py
'''
LeetCode Easy: 13. Roman to Integer
https://leetcode.com/problems/roman-to-integer/

Check RomanToInteger.cpp for brief solution explanation.
'''
class Solution:
    def romanToInt(self, s: str) -> int:
        legend = {'I':1, 'V': 5, 'X':10, 'L': 50, 'C': 100, 'D':500, 'M':1000}
        pre_value = 10000
        ans = 0
        for i in s:
            cur_value = legend[i]
            if pre_value<legend[i]:
                ans-=2*pre_value
            ans+=cur_value
            pre_value = cur_value
        return ans